### PR TITLE
Fixes #10484 - Clarify documentation about how to make a non-blocking…

### LINKED ISSUE #10484: Clarify documentation about how to make a non-blocking handler tree
**Jetty version(s)**
12.0.x

**Enhancement Description**
Since 12, the Core API allows writing entirely non-blocking handlers by extending `Handler.Abstract.NonBlocking` that are executed by the `AdaptiveExecutionStrategy`'s `ProduceConsume` sub-strategy.

Currently, the documentation mentions this fact but the information is scattered across many chapters. It also doesn't mention the fact that the entire handler tree has to be non-blocking, nor common pitfalls like the fact that `ContextHandlerCollection` is dynamic by default, which prevents it from being non-blocking.

This should be clarified.

### PR DESCRIPTION
**Pull request by sbordet**
… handler tree.

Added documentation about non-blocking Handlers.

### CODE REVIEW

File: documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-handler.adoc [hunk 1]
```diff
@@ -48,7 +48,38 @@ Refer to xref:pg-server-http-handler-use[this section] for more information abou
 You should write your own leaf ``Handler``s to implement your web application logic.
 Refer to xref:pg-server-http-handler-impl[this section] for more information about how to write your own ``Handler``s.
 
-// TODO: document ScopedHandler? Is this really necessary or just an implementation detail that application will never worry about?
+A leaf `Handler` may be declared as non-blocking (by extending `Handler.Abstract.NonBlocking`) or as blocking (by extending `Handler.Abstract`), to allow interaction with the xref:pg-arch-threads[Jetty threading architecture] for more efficient thread and CPU utilization during the request/response processing.
```
Comment (gregw) [inline documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-handler.adoc]: It is not just leaf handlers.   All handlers in a calling chain must be non-blocking

```suggestion
A `Handler` may be declared as non-blocking (by extending `Handler.Abstract.NonBlocking`) or as blocking (by extending `Handler.Abstract`), to allow interaction with the xref:pg-arch-threads[Jetty threading architecture] for more efficient thread and CPU utilization during the request/response processing.
```

**Review by gregw — DISMISSED** [review]
one tiny niggle...

File: documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-handler.adoc (hunk 1 above)
Comment (sbordet) [reply documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-handler.adoc]: @gregw yes, but `Handler.Wrapper` inherit the non-blocking-ness from their children, so it's important to define it only for leaf `Handler`s -- the others have good defaults.

Comment (gregw) [reply documentation/jetty-documentation/src/main/asciidoc/programming-guide/server/http/server-http-handler.adoc]: But you explain that below. So why start with something incorrect?

**Review by lorban — APPROVED** [review]

**Review by gregw — APPROVED** [review]

**Pull request merged**